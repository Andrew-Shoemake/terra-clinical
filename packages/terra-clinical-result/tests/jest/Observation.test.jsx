import React from 'react';
import Observation from '../../src/common/_Observation';
import DefaultResult from '../../src/terra-dev-site/test/clinical-result/TestResults';

// Snapshot Tests
describe('Observation', () => {
  it('should render a default Observation', () => {
    const nameHeaderCell = shallow(<Observation />);
    expect(nameHeaderCell).toMatchSnapshot();
  });

  it('should render an Observation with eventId', () => {
    const nameHeaderCell = shallow(<Observation eventId={DefaultResult.eventId} />);
    expect(nameHeaderCell).toMatchSnapshot();
  });

  it('should render an Observation with result', () => {
    const nameHeaderCell = shallow(<Observation result={DefaultResult.result} />);
    expect(nameHeaderCell).toMatchSnapshot();
  });

  it('should render an Observation with hideUnit', () => {
    const nameHeaderCell = shallow(<Observation hideUnit />);
    expect(nameHeaderCell).toMatchSnapshot();
  });

  describe('- Interpretaion', () => {
    it('should render an Observation with interpretation CRITICAL', () => {
      const nameHeaderCell = shallow(<Observation interpretation="CRITICAL" />);
      expect(nameHeaderCell).toMatchSnapshot();
    });

    it('should render an Observation with interpretation EXTREMEHIGH', () => {
      const nameHeaderCell = shallow(<Observation interpretation="EXTREMEHIGH" />);
      expect(nameHeaderCell).toMatchSnapshot();
    });

    it('should render an Observation with interpretation EXTREMELOW', () => {
      const nameHeaderCell = shallow(<Observation interpretation="EXTREMELOW" />);
      expect(nameHeaderCell).toMatchSnapshot();
    });

    it('should render an Observation with interpretation PANICHIGH', () => {
      const nameHeaderCell = shallow(<Observation interpretation="PANICHIGH" />);
      expect(nameHeaderCell).toMatchSnapshot();
    });

    it('should render an Observation with interpretation PANICLOW', () => {
      const nameHeaderCell = shallow(<Observation interpretation="PANICLOW" />);
      expect(nameHeaderCell).toMatchSnapshot();
    });

    it('should render an Observation with interpretation VABNORMAL', () => {
      const nameHeaderCell = shallow(<Observation interpretation="VABNORMAL" />);
      expect(nameHeaderCell).toMatchSnapshot();
    });

    it('should render an Observation with interpretation POSITIVE', () => {
      const nameHeaderCell = shallow(<Observation interpretation="POSITIVE" />);
      expect(nameHeaderCell).toMatchSnapshot();
    });

    it('should render an Observation with interpretation ABNORMAL', () => {
      const nameHeaderCell = shallow(<Observation interpretation="ABNORMAL" />);
      expect(nameHeaderCell).toMatchSnapshot();
    });

    it('should render an Observation with interpretation HIGH', () => {
      const nameHeaderCell = shallow(<Observation interpretation="HIGH" />);
      expect(nameHeaderCell).toMatchSnapshot();
    });

    it('should render an Observation with interpretation LOW', () => {
      const nameHeaderCell = shallow(<Observation interpretation="LOW" />);
      expect(nameHeaderCell).toMatchSnapshot();
    });

    it('should render an Observation with interpretation NORMAL', () => {
      const nameHeaderCell = shallow(<Observation interpretation="NORMAL" />);
      expect(nameHeaderCell).toMatchSnapshot();
    });

    it('should render an Observation with interpretation NEUTRAL', () => {
      const nameHeaderCell = shallow(<Observation interpretation="NEUTRAL" />);
      expect(nameHeaderCell).toMatchSnapshot();
    });
  });
});
