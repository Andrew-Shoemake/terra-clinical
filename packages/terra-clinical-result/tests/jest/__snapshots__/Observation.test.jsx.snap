// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Observation - Interpretaion should render an Observation with interpretation ABNORMAL 1`] = `<InjectIntl(ResultError) />`;

exports[`Observation - Interpretaion should render an Observation with interpretation CRITICAL 1`] = `<InjectIntl(ResultError) />`;

exports[`Observation - Interpretaion should render an Observation with interpretation EXTREMEHIGH 1`] = `<InjectIntl(ResultError) />`;

exports[`Observation - Interpretaion should render an Observation with interpretation EXTREMELOW 1`] = `<InjectIntl(ResultError) />`;

exports[`Observation - Interpretaion should render an Observation with interpretation HIGH 1`] = `<InjectIntl(ResultError) />`;

exports[`Observation - Interpretaion should render an Observation with interpretation LOW 1`] = `<InjectIntl(ResultError) />`;

exports[`Observation - Interpretaion should render an Observation with interpretation NEUTRAL 1`] = `<InjectIntl(ResultError) />`;

exports[`Observation - Interpretaion should render an Observation with interpretation NORMAL 1`] = `<InjectIntl(ResultError) />`;

exports[`Observation - Interpretaion should render an Observation with interpretation PANICHIGH 1`] = `<InjectIntl(ResultError) />`;

exports[`Observation - Interpretaion should render an Observation with interpretation PANICLOW 1`] = `<InjectIntl(ResultError) />`;

exports[`Observation - Interpretaion should render an Observation with interpretation POSITIVE 1`] = `<InjectIntl(ResultError) />`;

exports[`Observation - Interpretaion should render an Observation with interpretation VABNORMAL 1`] = `<InjectIntl(ResultError) />`;

exports[`Observation should render a default Observation 1`] = `<InjectIntl(ResultError) />`;

exports[`Observation should render an Observation with eventId 1`] = `<InjectIntl(ResultError) />`;

exports[`Observation should render an Observation with hideUnit 1`] = `<InjectIntl(ResultError) />`;

exports[`Observation should render an Observation with result 1`] = `
<Fragment>
  <span
    className="value"
  >
    12345.678
  </span>
  <span
    className="unit"
  >
    mL
  </span>
</Fragment>
`;
